app/pages/activity/components/recentActivities.tsx

```diff
@@ -78,50 +82,52 @@ const RecentActivities: React.FC<RecentActivitiesProps> = ({
               <span>{log.title}</span>
               <span className="text-muted-foreground">{log.duration}</span>
             </div>
-            <div className="flex items-center space-x-2">
-              <Popover>
-                <PopoverTrigger>
-                  <Button>Log Progress</Button>
-                </PopoverTrigger>
-                <PopoverContent className="bg-card p-4 rounded-lg shadow-lg max-w-xs mt-4">
-                  <div className="flex flex-col items-start space-y-2">
-                    <Input
-                      type="number"
-                      value={progressInputs[log.id] || 0}
-                      onChange={(e) =>
-                        handleProgressChange(log.id, Number(e.target.value))
-                      }
-                    />
-                    <div className="flex space-x-2">
-                      <Button onClick={() => handleAdd(log.id)}>Add</Button>
-                      <Button onClick={() => handleSubtract(log.id)}>
-                        Subtract
-                      </Button>
-                      <Button onClick={() => handleLogProgress(log.id)}>
-                        Log
-                      </Button>
+            {hasAccess(Number(userId), loggedUser) && (
+              <div className="flex items-center space-x-2">
+                <Popover>
+                  <PopoverTrigger>
```

Comment:
please instead of adding a lot of tabs create a new component , it wil lbe much easier to read then. applies for all other instances